test/testmultiply.jl

```diff
@@ -274,4 +274,69 @@ K3 = A ⊗ B ⊗ C
         @test (A ⊗ B ⊗ C) * v ≈ collect(A ⊗ B ⊗ C) * v
         @test (C ⊗ B ⊗ A) * v ≈ collect(C ⊗ B ⊗ A) * v
     end
+
+    @testset "diagonal kronecker product" begin
+        D1 = Diagonal(ones(3));
+        K = kronecker(D1, D1);
+        K3 = kronecker(D1, 3);
+        Kdense = kron(D1, D1);
+        K3dense = kron(D1, D1, D1);
+        @testset "with kronecker" begin
+            @test K * K == Kdense * Kdense
```

Comment:
Likely won't matter, but not safer here to use `≈`? One might change the test slightly later on.

Reply:
Actually maybe it'll be better to start with `Int` matrices anyway